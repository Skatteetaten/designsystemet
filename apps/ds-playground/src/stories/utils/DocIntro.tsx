import { ReactNode } from 'react';

import { Link } from '@skatteetaten/ds-buttons';
import { Paragraph } from '@skatteetaten/ds-typography';

interface DocIntroProps {
  name: string;
  url: string;
}

export const DocIntro = ({ name, url }: DocIntroProps): ReactNode => {
  return (
    <Paragraph variant={'ingress'}>
      {`Se `}
      <Link href={url}>{name}</Link>
      {` p√• stil og tone for eksempler og
        retningslinjer for denne komponenten.`}
    </Paragraph>
  );
};
