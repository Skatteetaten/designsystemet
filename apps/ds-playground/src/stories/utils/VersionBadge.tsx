import { ReactNode } from 'react';

interface VersionBadgeProps {
  packageName: string;
}

export const VersionBadge = ({ packageName }: VersionBadgeProps): ReactNode => {
  let color = 'green';

  let version =
    import.meta.env.DS_VERSIONS[`${packageName.replaceAll('-', '_')}`] ?? '..';
  if (import.meta.env.MODE === 'development') {
    color = 'red';
    version = 'developing';
  }
  if (import.meta.env['STORYBOOK_BUILD_VERSION']?.includes('SNAPSHOT')) {
    color = 'blue';
  }

  const normalizedPackageName = packageName.replaceAll('-', '--');
  const normalizedVersion = version.replaceAll('-', '--').replaceAll('_', '__');

  return (
    <img
      src={`https://img.shields.io/badge/%40skatteetaten%2Fds--${normalizedPackageName}-${normalizedVersion}-${color}?logo=npm`}
      alt={`@skatteetaten/ds-${packageName}@${version}`}
    />
  );
};
