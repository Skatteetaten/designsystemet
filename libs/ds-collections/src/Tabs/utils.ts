export const valueRegex = /^[A-Za-z][\w\-.:]*$/;
