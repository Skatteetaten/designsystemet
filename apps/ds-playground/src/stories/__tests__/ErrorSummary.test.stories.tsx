import { StoryFn, Meta, StoryObj } from '@storybook/react';
import { expect, userEvent, waitFor, within } from '@storybook/test';

import { headingAsArr } from '@skatteetaten/ds-core-utils';
import { ErrorSummary, TextField } from '@skatteetaten/ds-forms';
import { Paragraph } from '@skatteetaten/ds-typography';

import { loremIpsum } from './testUtils/storybook.testing.utils';
import { webComponent } from '../../../.storybook/webcomponent-decorator';

const meta = {
  component: ErrorSummary,
  title: 'Tester/ErrorSummary/ErrorSummary',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: {
      table: { disable: true },
      control: { disable: true },
    },
    content: {
      table: { disable: true },
      control: { disable: true },
    },
    shadowRootNode: {
      table: { disable: true },
    },
    showErrorSummary: {
      table: { disable: true },
    },
    title: {
      table: { disable: true },
    },
    titleAs: {
      options: [...headingAsArr],
      control: 'inline-radio',
      table: { disable: true },
    },
  },
} satisfies Meta<typeof ErrorSummary>;
export default meta;
type Story = StoryObj<typeof meta>;

const defaultChildren = [
  <ErrorSummary.Error key={'errorSummaryError_1'} referenceId={'id_1'}>
    {'Husk å fylle ut type varer'}
  </ErrorSummary.Error>,
  <ErrorSummary.Error key={'errorSummaryError_2'} referenceId={'id_2'}>
    {'Selger du varer og tjenester for egen regning?'}
  </ErrorSummary.Error>,
  false,
];

export const WithRef = {
  name: 'With Ref (FA1)',
  args: {
    ref: (instance: HTMLDivElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getAllByRole('generic')[1];
    await expect(container).toHaveAttribute('id', 'dummyIdForwardedFromRef');
  },
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes (FA2-5, B1)',
  args: {
    id: 'htmlId',
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
    showErrorSummary: true,
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getAllByRole('generic')[1];
    const errorSummary = canvas.getAllByRole('generic')[2];
    await expect(container).toHaveAttribute('id', 'htmlId');
    await expect(errorSummary).toHaveClass('dummyClassname');
    await expect(container).toHaveAttribute('lang', 'nb');
    await expect(container).toHaveAttribute('data-testid', '123ID');
  },
} satisfies Story;

export const Defaults = {
  name: 'Default (B2, B3 delvis)',
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getAllByRole('generic')[1];
    await expect(container).toBeInTheDocument();
    await expect(container).toHaveAttribute('aria-live', 'assertive');
    await expect(container).toHaveAttribute('aria-atomic');

    const errorSummary = container.querySelector('div');
    await expect(errorSummary).not.toBeInTheDocument();
  },
} satisfies Story;

export const WithChildren = {
  name: 'With Children (A1, A2, A3 delvis)',
  args: {
    showErrorSummary: true,
    children: defaultChildren,
  },
  argTypes: {
    children: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const listItems = canvas.getAllByRole('listitem');
    await expect(listItems).toHaveLength(2);
  },
} satisfies Story;

export const WithContent = {
  name: 'With Content (A1, A5, B1)',
  args: {
    showErrorSummary: true,
    title: 'Inntektsmottaker inneholder 2 feil:',
    content: <Paragraph>{loremIpsum}</Paragraph>,
  },
  argTypes: {
    content: { table: { disable: false } },
    title: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const icon = canvas.getByRole('img', { hidden: true });
    await expect(icon).toBeInTheDocument();
    await expect(icon).toHaveAttribute('aria-hidden', 'true');
  },
} satisfies Story;

export const WithChildrenAndContent = {
  name: 'With Children And Content (A1, A3 delvis, A5)',
  args: {
    showErrorSummary: true,
    children: defaultChildren,
    content: <Paragraph>{loremIpsum}</Paragraph>,
  },
  argTypes: {
    children: { table: { disable: false } },
    content: { table: { disable: false } },
  },
} satisfies Story;

export const WithChildrenAndContentMobile = {
  name: 'With Children And Content Mobile',
  args: {
    showErrorSummary: true,
    children: defaultChildren,
    content: <Paragraph>{loremIpsum}</Paragraph>,
  },
  argTypes: {
    children: { table: { disable: false } },
    content: { table: { disable: false } },
  },
  parameters: {
    viewport: {
      defaultViewport: '--mobile',
    },
  },
} satisfies Story;

export const WithChildrenAndContentBreakpointXs = {
  name: 'With Children And Content Breakpoint Xs',
  args: {
    showErrorSummary: true,
    children: defaultChildren,
    content: <Paragraph>{loremIpsum}</Paragraph>,
  },
  argTypes: {
    children: { table: { disable: false } },
    content: { table: { disable: false } },
  },
  parameters: {
    viewport: {
      defaultViewport: '--breakpoint-xs',
    },
  },
} satisfies Story;

export const WithTitleAs = {
  name: 'With TitleAs (A2)',
  args: {
    showErrorSummary: true,
    titleAs: 'h1',
    children: defaultChildren,
  },
  argTypes: {
    titleAs: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const heading = canvas.getByRole('heading', { level: 1 });
    await expect(heading).toBeInTheDocument();
  },
} satisfies Story;

const TemplateWithInput: StoryFn<typeof ErrorSummary> = () => (
  <>
    <TextField
      className={'bottomSpacingXL'}
      id={'input_aar'}
      label={'År'}
      value={1009}
      errorMessage={'Inntekståret må være etter 2008'}
      required
    />
    <ErrorSummary showErrorSummary>
      <ErrorSummary.Error referenceId={'input_aar'}>
        {'Inntekståret må være etter 2008'}
      </ErrorSummary.Error>
    </ErrorSummary>
  </>
);

export const WithInput = {
  render: TemplateWithInput,
  name: 'With Input (A4)',
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const link = canvas.getByRole('link');
    const input = canvas.getByLabelText('År');
    await userEvent.click(link);
    await waitFor(() => expect(input).toHaveFocus());
  },
} satisfies Story;

const TemplateWithShadowRootNode: StoryFn<typeof ErrorSummary> = () => {
  const element = document.querySelector('errorsummary-customelement');
  const shadowRoot = element?.shadowRoot;
  return (
    <div data-test-block>
      <TextField
        className={'bottomSpacingXL'}
        id={'input_aar'}
        label={'År'}
        value={1009}
        errorMessage={'Inntekståret må være etter 2008'}
        required
      />
      <ErrorSummary shadowRootNode={shadowRoot ?? undefined} showErrorSummary>
        <ErrorSummary.Error referenceId={'input_aar'}>
          {'Inntekståret må være etter 2008'}
        </ErrorSummary.Error>
      </ErrorSummary>
    </div>
  );
};

export const WithShadowRootNode = {
  render: TemplateWithShadowRootNode,
  name: 'With ShadowRootNode',
  decorators: [webComponent],
  parameters: {
    a11y: {
      //Stilen som slår av transitions og animations finnes ikke på innsiden av shadowDom
      //Slår derfor av denne slik at vi ikke får false positives
      disable: true,
    },
    imageSnapshot: {
      disable: true,
    },
    customElementName: 'errorsummary-customelement',
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    // errorLink finnes ikke utenfor shadowDom
    await expect(canvas.queryByRole('link')).not.toBeInTheDocument();

    const customElement = canvasElement.querySelector(
      `errorsummary-customelement`
    );
    await expect(customElement).toBeInTheDocument();
    const errorLink =
      customElement?.shadowRoot && customElement.shadowRoot.querySelector('a');
    await expect(errorLink).toBeInTheDocument();
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    await userEvent.click(errorLink!);
    const input =
      customElement?.shadowRoot &&
      customElement.shadowRoot.querySelector('input:focus');
    await expect(input).toBeInTheDocument();
  },
} satisfies Story;
