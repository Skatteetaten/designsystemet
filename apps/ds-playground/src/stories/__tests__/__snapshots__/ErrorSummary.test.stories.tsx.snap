// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/ErrorSummary/ErrorSummary Defaults play-test 1`] = `
<div data-test-block="true">
  <div class="_wrapper_1iu8h_1"
       aria-live="assertive"
       tabindex="-1"
       aria-atomic="true"
  >
  </div>
</div>
`;

exports[`Tester/ErrorSummary/ErrorSummary WithAttributes play-test 1`] = `
<div data-test-block="true">
  <div class="_wrapper_1iu8h_1"
       id="htmlId"
       lang="nb"
       data-testid="123ID"
       aria-live="assertive"
       tabindex="-1"
       aria-atomic="true"
  >
    <div class="_errorSummary_1iu8h_6 dummyClassname">
      <div class="_iconContainer_1iu8h_18">
        <svg class="_systemIcon_6zbcy_2 _systemIcon_large_6zbcy_13 _icon_1iu8h_18"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 14h-2v-4h2v4Zm0 4h-2v-2h2v2ZM1 21h22L12 2 1 21Z">
          </path>
        </svg>
      </div>
      <div class="_summaryContainer_1iu8h_43">
        <h2 class="_heading_12p5o_1 _heading_level4_12p5o_43 _heading_hasSpacingSmall_12p5o_54">
          For å gå videre må du rette opp 0 feil:
        </h2>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/ErrorSummary/ErrorSummary WithChildren play-test 1`] = `
<div data-test-block="true">
  <div class="_wrapper_1iu8h_1"
       aria-live="assertive"
       tabindex="-1"
       aria-atomic="true"
  >
    <div class="_errorSummary_1iu8h_6">
      <div class="_iconContainer_1iu8h_18">
        <svg class="_systemIcon_6zbcy_2 _systemIcon_large_6zbcy_13 _icon_1iu8h_18"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 14h-2v-4h2v4Zm0 4h-2v-2h2v2ZM1 21h22L12 2 1 21Z">
          </path>
        </svg>
      </div>
      <div class="_summaryContainer_1iu8h_43">
        <h2 class="_heading_12p5o_1 _heading_level4_12p5o_43 _heading_hasSpacingSmall_12p5o_54">
          For å gå videre må du rette opp 2 feil:
        </h2>
        <ul class="_linkGroup_w23nb_1">
          <li class="_linkGroupItem_w23nb_8">
            <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
              </path>
            </svg>
            <a class="_link_z7jba_1"
               href="#id_1"
            >
              Husk å fylle ut type varer
            </a>
          </li>
          <li class="_linkGroupItem_w23nb_8">
            <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
              </path>
            </svg>
            <a class="_link_z7jba_1"
               href="#id_2"
            >
              Selger du varer og tjenester for egen regning?
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/ErrorSummary/ErrorSummary WithChildrenAndContent smoke-test 1`] = `
<div data-test-block="true">
  <div class="_wrapper_1iu8h_1"
       aria-live="assertive"
       tabindex="-1"
       aria-atomic="true"
  >
    <div class="_errorSummary_1iu8h_6">
      <div class="_iconContainer_1iu8h_18">
        <svg class="_systemIcon_6zbcy_2 _systemIcon_large_6zbcy_13 _icon_1iu8h_18"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 14h-2v-4h2v4Zm0 4h-2v-2h2v2ZM1 21h22L12 2 1 21Z">
          </path>
        </svg>
      </div>
      <div class="_summaryContainer_1iu8h_43">
        <h2 class="_heading_12p5o_1 _heading_level4_12p5o_43 _heading_hasSpacingSmall_12p5o_54">
          For å gå videre må du rette opp 2 feil:
        </h2>
        <ul class="_linkGroup_w23nb_1">
          <li class="_linkGroupItem_w23nb_8">
            <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
              </path>
            </svg>
            <a class="_link_z7jba_1"
               href="#id_1"
            >
              Husk å fylle ut type varer
            </a>
          </li>
          <li class="_linkGroupItem_w23nb_8">
            <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
              </path>
            </svg>
            <a class="_link_z7jba_1"
               href="#id_2"
            >
              Selger du varer og tjenester for egen regning?
            </a>
          </li>
        </ul>
        <div class="_content_1iu8h_57">
          <p class="_paragraph_1kzt5_1 _paragraph_standard_1kzt5_17">
            Lorem ipsum dolor sit amet. Alle som har laget en nettside, trengt litt fylltekst eller bare surfet rundt på nettet har antageligvis sett disse ordene, etterfulgt av en tilsynelatende eviglang tekst fylt med latinske liksomsetninger.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/ErrorSummary/ErrorSummary WithChildrenAndContentBreakpointXs smoke-test 1`] = `
<div data-test-block="true">
  <div class="_wrapper_1iu8h_1"
       aria-live="assertive"
       tabindex="-1"
       aria-atomic="true"
  >
    <div class="_errorSummary_1iu8h_6">
      <div class="_iconContainer_1iu8h_18">
        <svg class="_systemIcon_6zbcy_2 _systemIcon_large_6zbcy_13 _icon_1iu8h_18"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 14h-2v-4h2v4Zm0 4h-2v-2h2v2ZM1 21h22L12 2 1 21Z">
          </path>
        </svg>
      </div>
      <div class="_summaryContainer_1iu8h_43">
        <h2 class="_heading_12p5o_1 _heading_level4_12p5o_43 _heading_hasSpacingSmall_12p5o_54">
          For å gå videre må du rette opp 2 feil:
        </h2>
        <ul class="_linkGroup_w23nb_1">
          <li class="_linkGroupItem_w23nb_8">
            <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
              </path>
            </svg>
            <a class="_link_z7jba_1"
               href="#id_1"
            >
              Husk å fylle ut type varer
            </a>
          </li>
          <li class="_linkGroupItem_w23nb_8">
            <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
              </path>
            </svg>
            <a class="_link_z7jba_1"
               href="#id_2"
            >
              Selger du varer og tjenester for egen regning?
            </a>
          </li>
        </ul>
        <div class="_content_1iu8h_57">
          <p class="_paragraph_1kzt5_1 _paragraph_standard_1kzt5_17">
            Lorem ipsum dolor sit amet. Alle som har laget en nettside, trengt litt fylltekst eller bare surfet rundt på nettet har antageligvis sett disse ordene, etterfulgt av en tilsynelatende eviglang tekst fylt med latinske liksomsetninger.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/ErrorSummary/ErrorSummary WithChildrenAndContentMobile smoke-test 1`] = `
<div data-test-block="true">
  <div class="_wrapper_1iu8h_1"
       aria-live="assertive"
       tabindex="-1"
       aria-atomic="true"
  >
    <div class="_errorSummary_1iu8h_6">
      <div class="_iconContainer_1iu8h_18">
        <svg class="_systemIcon_6zbcy_2 _systemIcon_large_6zbcy_13 _icon_1iu8h_18"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 14h-2v-4h2v4Zm0 4h-2v-2h2v2ZM1 21h22L12 2 1 21Z">
          </path>
        </svg>
      </div>
      <div class="_summaryContainer_1iu8h_43">
        <h2 class="_heading_12p5o_1 _heading_level4_12p5o_43 _heading_hasSpacingSmall_12p5o_54">
          For å gå videre må du rette opp 2 feil:
        </h2>
        <ul class="_linkGroup_w23nb_1">
          <li class="_linkGroupItem_w23nb_8">
            <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
              </path>
            </svg>
            <a class="_link_z7jba_1"
               href="#id_1"
            >
              Husk å fylle ut type varer
            </a>
          </li>
          <li class="_linkGroupItem_w23nb_8">
            <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
              </path>
            </svg>
            <a class="_link_z7jba_1"
               href="#id_2"
            >
              Selger du varer og tjenester for egen regning?
            </a>
          </li>
        </ul>
        <div class="_content_1iu8h_57">
          <p class="_paragraph_1kzt5_1 _paragraph_standard_1kzt5_17">
            Lorem ipsum dolor sit amet. Alle som har laget en nettside, trengt litt fylltekst eller bare surfet rundt på nettet har antageligvis sett disse ordene, etterfulgt av en tilsynelatende eviglang tekst fylt med latinske liksomsetninger.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/ErrorSummary/ErrorSummary WithContent play-test 1`] = `
<div data-test-block="true">
  <div class="_wrapper_1iu8h_1"
       aria-live="assertive"
       tabindex="-1"
       aria-atomic="true"
  >
    <div class="_errorSummary_1iu8h_6">
      <div class="_iconContainer_1iu8h_18">
        <svg class="_systemIcon_6zbcy_2 _systemIcon_large_6zbcy_13 _icon_1iu8h_18"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 14h-2v-4h2v4Zm0 4h-2v-2h2v2ZM1 21h22L12 2 1 21Z">
          </path>
        </svg>
      </div>
      <div class="_summaryContainer_1iu8h_43">
        <h2 class="_heading_12p5o_1 _heading_level4_12p5o_43 _heading_hasSpacingSmall_12p5o_54">
          Inntektsmottaker inneholder 2 feil:
        </h2>
        <div class="_content_1iu8h_57 _contentWithoutChildren_1iu8h_60">
          <p class="_paragraph_1kzt5_1 _paragraph_standard_1kzt5_17">
            Lorem ipsum dolor sit amet. Alle som har laget en nettside, trengt litt fylltekst eller bare surfet rundt på nettet har antageligvis sett disse ordene, etterfulgt av en tilsynelatende eviglang tekst fylt med latinske liksomsetninger.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/ErrorSummary/ErrorSummary WithFocus play-test 1`] = `
<div data-test-block="true">
  <div class>
    <label id="labelId-:rv:"
           class="_label_348na_5"
           for="input_aar"
    >
      År
    </label>
    <input id="input_aar"
           class="_textbox_3tcss_5    _textboxMarginTop_3tcss_21"
           autocomplete="off"
           required
           aria-describedby="textFieldErrorId-:rt:"
           aria-invalid="true"
           value="1009"
    >
    <div id="textFieldErrorId-:rt:"
         aria-live="assertive"
         aria-atomic="true"
    >
      <div class="_errorMessage_1kdf4_15">
        <svg class="_systemIcon_6zbcy_2 _systemIcon_small_6zbcy_5 _errorMessageIcon_1kdf4_25"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 14h-2v-4h2v4Zm0 4h-2v-2h2v2ZM1 21h22L12 2 1 21Z">
          </path>
        </svg>
        <p class="_errorMessageText_1kdf4_30">
          Inntekståret må være etter 2008
        </p>
      </div>
    </div>
  </div>
  <div class="_wrapper_1iu8h_1"
       id="errorSummary1"
       aria-live="assertive"
       tabindex="-1"
       aria-atomic="true"
  >
    <div class="_errorSummary_1iu8h_6">
      <div class="_iconContainer_1iu8h_18">
        <svg class="_systemIcon_6zbcy_2 _systemIcon_large_6zbcy_13 _icon_1iu8h_18"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 14h-2v-4h2v4Zm0 4h-2v-2h2v2ZM1 21h22L12 2 1 21Z">
          </path>
        </svg>
      </div>
      <div class="_summaryContainer_1iu8h_43">
        <h2 class="_heading_12p5o_1 _heading_level4_12p5o_43 _heading_hasSpacingSmall_12p5o_54">
          For å gå videre må du rette opp 1 feil:
        </h2>
        <ul class="_linkGroup_w23nb_1">
          <li class="_linkGroupItem_w23nb_8">
            <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
              </path>
            </svg>
            <a class="_link_z7jba_1"
               href="#input_aar"
            >
              Inntekståret må være etter 2008
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <button class="_button_6ybyo_1 _button_primary_6ybyo_1   topSpacingXL"
          type="button"
  >
    <span class="_buttonText_6ybyo_69">
      Send
    </span>
  </button>
</div>
`;

exports[`Tester/ErrorSummary/ErrorSummary WithInput play-test 1`] = `
<div data-test-block="true">
  <div class="bottomSpacingXL">
    <label id="labelId-:rj:"
           class="_label_348na_5"
           for="input_aar"
    >
      År
    </label>
    <input id="input_aar"
           class="_textbox_3tcss_5    _textboxMarginTop_3tcss_21"
           autocomplete="off"
           required
           aria-describedby="textFieldErrorId-:rh:"
           aria-invalid="true"
           value="1009"
    >
    <div id="textFieldErrorId-:rh:"
         aria-live="assertive"
         aria-atomic="true"
    >
      <div class="_errorMessage_1kdf4_15">
        <svg class="_systemIcon_6zbcy_2 _systemIcon_small_6zbcy_5 _errorMessageIcon_1kdf4_25"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 14h-2v-4h2v4Zm0 4h-2v-2h2v2ZM1 21h22L12 2 1 21Z">
          </path>
        </svg>
        <p class="_errorMessageText_1kdf4_30">
          Inntekståret må være etter 2008
        </p>
      </div>
    </div>
  </div>
  <div class="_wrapper_1iu8h_1"
       aria-live="assertive"
       tabindex="-1"
       aria-atomic="true"
  >
    <div class="_errorSummary_1iu8h_6">
      <div class="_iconContainer_1iu8h_18">
        <svg class="_systemIcon_6zbcy_2 _systemIcon_large_6zbcy_13 _icon_1iu8h_18"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 14h-2v-4h2v4Zm0 4h-2v-2h2v2ZM1 21h22L12 2 1 21Z">
          </path>
        </svg>
      </div>
      <div class="_summaryContainer_1iu8h_43">
        <h2 class="_heading_12p5o_1 _heading_level4_12p5o_43 _heading_hasSpacingSmall_12p5o_54">
          For å gå videre må du rette opp 1 feil:
        </h2>
        <ul class="_linkGroup_w23nb_1">
          <li class="_linkGroupItem_w23nb_8">
            <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
              </path>
            </svg>
            <a class="_link_z7jba_1"
               href="#input_aar"
            >
              Inntekståret må være etter 2008
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/ErrorSummary/ErrorSummary WithRef play-test 1`] = `
<div data-test-block="true">
  <div class="_wrapper_1iu8h_1"
       aria-live="assertive"
       tabindex="-1"
       aria-atomic="true"
       id="dummyIdForwardedFromRef"
  >
  </div>
</div>
`;

exports[`Tester/ErrorSummary/ErrorSummary WithShadowRootNode play-test 1`] = `
<div data-test-block="true">
  <errorsummary-customelement>
  </errorsummary-customelement>
</div>
`;

exports[`Tester/ErrorSummary/ErrorSummary WithTitleAs play-test 1`] = `
<div data-test-block="true">
  <div class="_wrapper_1iu8h_1"
       aria-live="assertive"
       tabindex="-1"
       aria-atomic="true"
  >
    <div class="_errorSummary_1iu8h_6">
      <div class="_iconContainer_1iu8h_18">
        <svg class="_systemIcon_6zbcy_2 _systemIcon_large_6zbcy_13 _icon_1iu8h_18"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 14h-2v-4h2v4Zm0 4h-2v-2h2v2ZM1 21h22L12 2 1 21Z">
          </path>
        </svg>
      </div>
      <div class="_summaryContainer_1iu8h_43">
        <h1 class="_heading_12p5o_1 _heading_level4_12p5o_43 _heading_hasSpacingSmall_12p5o_54">
          For å gå videre må du rette opp 2 feil:
        </h1>
        <ul class="_linkGroup_w23nb_1">
          <li class="_linkGroupItem_w23nb_8">
            <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
              </path>
            </svg>
            <a class="_link_z7jba_1"
               href="#id_1"
            >
              Husk å fylle ut type varer
            </a>
          </li>
          <li class="_linkGroupItem_w23nb_8">
            <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26"
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
              </path>
            </svg>
            <a class="_link_z7jba_1"
               href="#id_2"
            >
              Selger du varer og tjenester for egen regning?
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
